hudi-common/src/main/avro/HoodieCompactionOperation.avsc

```diff
@@ -41,7 +41,7 @@
                      "default": null
                   },
                   {
-                     "name":"dataFilePath",
+                     "name":"baseFilePath",
```

Comment:
@pratyakshsharma : Instead of renaming fields, add an alias ?

http://bigdatafindings.blogspot.com/2016/05/schema-evolution-with-avro.html

Reply:
@pratyakshsharma @bvaradar is there a test here that checks for backwards compatibility?  i.e we should be very sure that existing plans should be readable with new schema.. (or may be also new plans by old schema?) 

This is my main concern. Otherwise, renaming is not worth the hassle. :) for this particular case.. thoughts?

Reply:
@pratyakshsharma : Alias should be the new name "baseFilePath". We need to keep the name unchanged here. I agree with @vinothchandar . We need to be very careful when changing serialized metadata entries. This includes
1. Avro Model (*avsc )
2. Serialized Model Classes - org.apache.hudi.common.model.HoodieCommitMetadata (not relevant for this change)

Can you take a look at TestHoodieCommitMetadata to see an example of how we test backwards compatibility. We would need to do something similar for this change.

Reply:
Checking this.

Reply:
@bvaradar We are basically adding a new adding a new field in TestHoodieCommitMetadata class and then testing compatibility. However in this PR, I am not sure of how an avro record with field "baseFilePath" will be converted to a POJO with field dataFilePath while deserializing. I have removed the alias for now. If you can guide me here, that would be useful.